parser/prism/Translator.cc

```diff
@@ -980,6 +972,29 @@ std::unique_ptr<parser::Node> Translator::translateStatements(pm_statements_node
     return make_unique<parser::Begin>(parser.translateLocation(&stmtsNode->base.location), std::move(sorbetStmts));
 }
 
+std::unique_ptr<parser::Node> Translator::translateConstantPath(pm_constant_path_node *node, bool isAssignment) {
+    pm_location_t *loc = &node->base.location;
+
+    std::string_view name = parser.resolveConstant(node->name);
+
+    std::unique_ptr<parser::Node> parent;
+    if (node->parent) {
+        // This constant reference is chained onto another constant reference.
+        // E.g. if `node` is pointing to `B`, then then `A` is the `parent` in `A::B::C`.
+        parent = translate(node->parent);
+    } else {                                                // This is a fully qualified constant reference, like `::A`.
+        pm_location_t *delimiterLoc = &node->delimiter_loc; // The location of the `::`
+        parent = make_unique<parser::Cbase>(parser.translateLocation(delimiterLoc));
+    }
+
+    if (isAssignment) {
+        return make_unique<parser::ConstLhs>(parser.translateLocation(loc), std::move(parent),
+                                             gs.enterNameConstant(name));
+    } else {
+        return make_unique<parser::Const>(parser.translateLocation(loc), std::move(parent), gs.enterNameConstant(name));
+    }
```

Comment:
Here is where we make different Sorbet nodes based on the context of the constant path node.